### PR TITLE
[Refactor] [Step 1/2] Change the name from master to leader

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Change the master related code to leader.
Split the task into two steps: 
1. change the name from master to leader in the internal code paths.
2. change the name from master to leader in the interface.

The advantage of this is that we only need to rollback the step 2 if If there are compatibility problems.

This is PR is still the step 1 of the task.
The previous work is done by https://github.com/StarRocks/starrocks/pull/8787

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 37 (64.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowProcAction.java|0|1|00.00%|[80]|
|:large_blue_circle:|com/starrocks/http/meta/GlobalDictMetaService.java|0|1|00.00%|[47]|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|1|00.00%|[134]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|1|00.00%|[73]|
|:large_blue_circle:|com/starrocks/task/ExportExportingTask.java|0|1|00.00%|[93]|
|:large_blue_circle:|com/starrocks/leader/Checkpoint.java|0|1|00.00%|[201]|
|:large_blue_circle:|com/starrocks/http/meta/ColocateMetaService.java|0|1|00.00%|[99]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|3|66.67%|[1095]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|5|7|71.43%|[373, 578]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|6|8|75.00%|[126, 141]|
|:large_blue_circle:|com/starrocks/analysis/RedirectStatus.java|3|4|75.00%|[25]|
|:large_blue_circle:|com/starrocks/task/LeaderTaskExecutor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/LeaderTask.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/http/HttpServer.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -316,7 +316,7 @@ public class Config extends ConfigBase {
     public static int bdbje_lock_timeout_second = 1;
 
     /**
-     * Set the maximum acceptable clock skew between non-master FE to Master FE host.
+     * Set the maximum acceptable clock skew between non-leader FE to Leader FE host.
      * This value is checked whenever a non-master FE establishes a connection to master FE via BDBJE.
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:320]: non-leader
to leader

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 2]
```diff
@@ -399,19 +399,19 @@ public class Config extends ConfigBase {
...
      * to another node and set this config to true to try to restart the FE.
      */
     @ConfField
     public static String metadata_failure_recovery = "false";
 
     /**
-     * If true, non-master FE will ignore the meta data delay gap between Master FE and its self,
+     * If true, non-master FE will ignore the meta data delay gap between Leader FE and its self,
      * even if the metadata delay gap exceeds *meta_delay_toleration_second*.
      * Non-master FE will still offer read service.
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:412]: Non-leader

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -399,19 +399,19 @@ public class Config extends ConfigBase {
...
      * <p>
-     * This is helpful when you try to stop the Master FE for a relatively long time for some reason,
+     * This is helpful when you try to stop the Leader FE for a relatively long time for some reason,
      * but still wish the non-master FE can offer read service.
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:415]: non-leader

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: done

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: done

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java (hunk 2 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 24 / 37 (64.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/rest/ShowProcAction.java|0|1|00.00%|[80]|
|:large_blue_circle:|com/starrocks/http/meta/GlobalDictMetaService.java|0|1|00.00%|[47]|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|1|00.00%|[134]|
|:large_blue_circle:|com/starrocks/task/ExportPendingTask.java|0|1|00.00%|[73]|
|:large_blue_circle:|com/starrocks/task/ExportExportingTask.java|0|1|00.00%|[93]|
|:large_blue_circle:|com/starrocks/leader/Checkpoint.java|0|1|00.00%|[201]|
|:large_blue_circle:|com/starrocks/http/meta/ColocateMetaService.java|0|1|00.00%|[99]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|2|3|66.67%|[1095]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|5|7|71.43%|[373, 578]|
|:large_blue_circle:|com/starrocks/load/ExportChecker.java|6|8|75.00%|[126, 141]|
|:large_blue_circle:|com/starrocks/analysis/RedirectStatus.java|3|4|75.00%|[25]|
|:large_blue_circle:|com/starrocks/task/LeaderTaskExecutor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/task/LeaderTask.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/http/HttpServer.java|1|1|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Review by padmejin — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Pull request merged**